tests/jsmarionette/client/marionette-client/test/marionette/client-test.js

```diff
@@ -1117,21 +1148,57 @@ suite('marionette/client', function() {
       test('should call _executeScript', function() {
         assert.deepEqual(calledWith, [{
           name: 'executeJSScript',
-          parameters: {script: script, timeout: true, args: null}
+          parameters: {
+            script: script,
+            timeout: true,
+            args: null,
+            sandbox: 'simpletest',
+          }
         }, commandCallback]);
       });
     });
 
     suite('.executeAsyncScript', function() {
-      setup(function() {
+      test('should call _executeScript', function() {
         subject.executeAsyncScript(script, commandCallback);
+        assert.deepEqual(calledWith, [
+          {
+            name: 'executeAsyncScript',
+            parameters: {
+              script: script,
+              args: null,
+              sandbox: 'default',
+            }
+          },
+          commandCallback
+        ]);
       });
 
-      test('should call _executeScript', function() {
+      test('should call _executeScript with arguments', function() {
+        subject.executeAsyncScript(script, ['foo'], commandCallback);
+        assert.deepEqual(calledWith, [
+          {
+            name: 'executeAsyncScript',
+            parameters: {
+              script: script,
+              args: ['foo'],
+              sandbox: 'default',
+            }
+          },
+          commandCallback
+        ]);
+      });
+
+      test('should call _executeScript with custom sandbox', function() {
+        subject.executeAsyncScript(script, null, commandCallback, 'foo');
         assert.deepEqual(calledWith, [
           {
             name: 'executeAsyncScript',
-            parameters: {script: script, args: null}
+            parameters: {
+              script: script,
+              args: null,
+              sandbox: 'foo',
+            }
```

Comment:
We also probably need a test for executeJSScript that makes sure `_executeScript` is called with the simpletest sandbox, so we don't break that in the future.